### PR TITLE
Feature/track as function

### PR DESCRIPTION
**Pull request by adiwegs**
Hi

Thanks for your work! I have been looking into this too.  #77 

I was trying to figure out a way to handle the tracking with Javascript. With this, we don't need to reload the page after the user has accepted the cookie and we keep the custom attributes handling functionality. I think we can simply enclose the tracking in a function and then call it when the user has accepted:

```
function wpmautic_send(){
    mt('send', 'pageview'<?php echo count( $attrs ) > 0 ? ', ' . wp_json_encode( $attrs ) : ''; ?>);
}
```

Here is a possible integration with AmauriC/tarteaucitron.js:
```
// wpmautic
	tarteaucitron.services.wpmautic = {
		"key": "wpmautic",
		"type": "analytic",
		"name": "WP Mautic",
		"uri": "https://www.mautic.org/privacy-policy/",
		"needConsent": true,
		"cookies": ['mtc_id', 'mtc_sid'],
		"js": function () {
			"use strict";
			if (tarteaucitron.user.mauticurl === undefined) {
				return;
			}

			window['MauticTrackingObject'] = 'mt';
			window['mt'] = window['mt'] || function() {
				(window['mt'].q = window['mt'].q || []).push(arguments);
			};

			tarteaucitron.addScript(tarteaucitron.user.mauticurl, '', function() {
				wpmautic_send();
			});
		},
		"fallback": function () {
			"use strict";
			// when use deny cookie
		}
};
```
Details in the PR

### CODE REVIEW

**Comment by shulard** [comment]
Hello !

Sorry about multiple comments, I've got some trouble with the Github Review system ^^.

This is an interesting and tiny change. I'm not against working with a JavaScript function here instead of exposing the code directly.

I think that you must rebase your PR on master because the 1b680a0 commit is already on master and is also in your branch.

I think we can also add some details in the plugin page about that change. Maybe describing how to load and call the Mautic tracker manually ?

**Comment by adiwegs** [comment]
Hi @shulard, thanks for your review. I did some improvements based on your comment.

And I also added my implementation of tarteaucitron to the plugin. It is just a start and only uses the basic options. But it is enough to get a GDPR banner/popup.

image

I'm not sure if this plugin is the best place for this though. Maybe it is too niche and should be an extra plugin?

I guess we need to do a second PR to merge #77 again to the master branch.

Any help with the tests would be appreciated. 🙏

**Comment by shulard** [comment]
Hello !

I'm not sure that including the "tarteaucitron" tool is the goal of the mautic plugin. We need to ensure that everything is available for the users to be GDPR compliant but providing a full implementation seems a bit too much for me.

Maybe we can work on such things in a specific PR but it must be discussed with wider Mautic community to get more feedbacks. I think this PR goal is to provide a JavaScript function to be called from the frontend side and execute the tracker when possible.

Can you rework your commits to let just those changes ?

I can help on unit tests in during the next week.

Thank you for your work 👍.

**Comment by adiwegs** [comment]
@shulard I agree. Therefore, I removed tarteaucitron and will create a new branch/PR for that discussion.

File: wpmautic.php [hunk 1]
```diff
@@ -145,19 +150,40 @@ function wpmautic_inject_script() {
...
 	})(window,document,'script','<?php echo esc_url( $base_url ); ?>','mt');
-
-	mt('send', 'pageview'<?php echo count( $attrs ) > 0 ? ', ' . wp_json_encode( $attrs ) : ''; ?>);
 </script>
 	<?php
 }
+/**
+ * Add the mt('send', 'pageview') script with optional tracking attributes
+ */
+function wpmautic_inject_send_script() {
```
Comment (shulard) [inline wpmautic.php]: I think this method can be merged in the `wpmautic_inject_script` one. We can still create the JavaScript function but it'll just create one `<script>` tag in the rendered page.

**Pull request closed without merging**